File: Data Structures & Algorithms/search-2d-matrix/submission-0.java
class Solution {
    public boolean searchMatrix(int[][] arr, int target) {
        int row = arr.length;
        int col = arr[0].length;
        int low = 0;
        int high = row*col-1;
        while(low<=high){
            int mid = low+(high-low)/2;
            if(arr[mid/row][mid%row]>target){
                high = mid-1;
            }
            else if(arr[mid/row][mid%row]<target){
                low = mid+1;
            }
            else{
                return true;
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/find-missing-and-repeated-values/submission-0.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        row = len(grid)
        col = len(grid[0])
        seen = set()
        l = []
        mp = {}
        for i in range(row):
            for j in range(col):
                seen.add(grid[i][j])
                mp[grid[i][j]] = mp.get(grid[i][j],0)+1
        
        for i in mp:
            if mp.get(i)>1:
                l.append(i)
        
        

        for i in range(1,row*col+1):
            if i not in seen:
                l.append(i)
        
        return l

